runtime/rasdump/dmpsup.c

```diff
@@ -1365,7 +1366,72 @@ initSystemInfo(J9JavaVM *vm)
 			}
 		}
 	}
-	appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+	{
+		J9RASSystemInfo* corePatternInfo = appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+		if ( corePatternInfo != NULL ) {
+			/* A common core_pattern is Dynatrace; for example, |/opt/dynatrace/oneagent/agent/rdp
+			 * This program sends the core to the originally configured
+			 * core_pattern as storied in, for example,
+			 * /opt/dynatrace/oneagent/agent/conf/original_core_pattern
+			 *
+			 * First, check if the core_pattern is Dynatrace. If so, extract its
+			 * installation directory and then read original_core_pattern
+			 * relative to that.
+			 */
+			const char *corePattern = (const char *)corePatternInfo->data;
+			if ( corePattern[0] == '|' ) {
+				char* dynatracePath = strstr(corePattern, "/oneagent/agent/rdp");
+				J9RAS* rasStruct = vm->j9ras;
+
+				/* Check if core_pattern includes the Dynatrace agent,
+				 * and sanity check that the length of core_pattern is smaller than our buffer.
+				 */
+				if ( dynatracePath != NULL && strlen(corePattern) < MAX_INTERESTING_LENGTH && rasStruct != NULL ) {
+					char namebuf[MAX_INTERESTING_LENGTH];
+					/* The length of the agent path prefix, minus the pipe character. */
+					size_t prefixLength = dynatracePath - corePattern - 1;
+					IDATA fd = -1;
+
+					/* Copy starting after the pipe character. */
+					strncpy(namebuf, corePattern + 1, prefixLength);
+					namebuf[prefixLength] = '\0';
+
+					/* Append the location of original_core_pattern. */
+					strncat(namebuf, "/oneagent/agent/conf/original_core_pattern", sizeof(namebuf) - prefixLength - 1);
+
+					/* Finally, read the file contents. */
+					fd = j9file_open(namebuf, EsOpenRead, 0);
+					if (fd != -1) {
+						char buf[MAX_INTERESTING_LENGTH];
+						char* read = j9file_read_text(fd, buf, sizeof(buf));
+						if( read == &buf[0] ) {
+							size_t bufLen = 0;
+							J9RASSystemInfo* newSystemInfo = NULL;
+
+							/* Make sure the string is only one line and null terminated. */
```

Comment:
This comment is misleading (if the string is long, it won't be null terminated).

Reply:
Ahh, yes, I see. In that case, it'll be null terminated by the memset below.